pkg/workerapi/workerserver/artifact.go

```diff
@@ -0,0 +1,40 @@
+package workerserver
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/iver-wharf/wharf-core/pkg/ginutil"
+)
+
+type artifactModule struct {
+	*restServer
+}
```

Comment:
This form of embedding is a little bit hard to read IMO. As you're only referencing the `artifactReader` I think it'd make the intent clearer of what you actually use the `restServer` for if you only explicitly had that inside this `artifactModule` struct, like so:

```go
type artifactModule struct {
	artifactReader ArtifactReader
}
```